projects/npm-tools/packages/npm-scripts/test/sass/build.js

```diff
@@ -0,0 +1,262 @@
+
+		expect(fs.existsSync(mainScssFilePath)).toBe(false);
+
+		buildSass(path.join(FIXTURES, 'modified'), {
+			outputDir: tempDir,
+		});
+
+		expect(fs.existsSync(mainScssFilePath)).toBe(true);
+
+		const fileCreationTime1 = fs.statSync(mainScssFilePath).ctime.getTime();
+
+		buildSass(path.join(FIXTURES, 'modified'), {
+			outputDir: tempDir,
+		});
+
+		const fileCreationTime2 = fs.statSync(mainScssFilePath).ctime.getTime();
+
+		expect(fileCreationTime2).toBe(fileCreationTime1);
+
+		fs.unlinkSync(mainScssFilePath);
+
+		buildSass(path.join(FIXTURES, 'modified'), {
+			outputDir: tempDir,
+		});
+
+		const fileCreationTime3 = fs.statSync(mainScssFilePath).ctime.getTime();
+
+		expect(fileCreationTime3).toBeGreaterThan(fileCreationTime1);
+		expect(fileCreationTime3).toBeGreaterThan(fileCreationTime2);
+	});
+
+	it('rebuilds sass if partial has been modified', () => {
+		const mainScssFilePath = path.join(tempDir, 'partial.scss');
+
+		expect(fs.existsSync(mainScssFilePath)).toBe(false);
+
+		buildSass(path.join(FIXTURES, 'partial'), {
+			outputDir: tempDir,
+		});
+
+		expect(fs.existsSync(mainScssFilePath)).toBe(true);
+
+		const fileCreationTime1 = fs.statSync(mainScssFilePath).ctime.getTime();
+
+		buildSass(path.join(FIXTURES, 'partial'), {
+			outputDir: tempDir,
+		});
+
+		const fileCreationTime2 = fs.statSync(mainScssFilePath).ctime.getTime();
+
+		expect(fileCreationTime2).toBe(fileCreationTime1);
+
+		fs.unlinkSync(path.join(tempDir, '_some-partial.scss'));
+
+		buildSass(path.join(FIXTURES, 'partial'), {
+			outputDir: tempDir,
+		});
+
+		const fileCreationTime3 = fs.statSync(mainScssFilePath).ctime.getTime();
+
+		expect(fileCreationTime3).toBeGreaterThan(fileCreationTime1);
+		expect(fileCreationTime3).toBeGreaterThan(fileCreationTime2);
+	});
+
+	it('logs message if no files found', () => {
+		expect(log).toBeCalledTimes(0);
+
+		buildSass(path.join(FIXTURES, 'none'), {
+			outputDir: tempDir,
+		});
+
+		expect(log).toBeCalledTimes(1);
+
+		expect(log).toBeCalledWith('BUILD CSS: No files found.');
+	});
+
+	it('builds css with partials', () => {
+		buildSass(path.join(FIXTURES, 'partial'), {
+			outputDir: tempDir,
+		});
+
+		expect(fs.existsSync(path.join(tempDir, '_some-partial.scss'))).toBe(
+			true
+		);
+
+		expect(fs.existsSync(path.join(tempSassDir, 'partial.css'))).toBe(true);
+		expect(fs.existsSync(path.join(tempSassDir, 'partial.css.map'))).toBe(
+			true
+		);
+
+		expect(fs.readFileSync(path.join(tempSassDir, 'partial.css'), 'utf-8'))
+			.toMatchInlineSnapshot(`
+		".this-is-a-partial {
+		  color: blue;
+		}
+
+		/*# sourceMappingURL=partial.css.map */"
+	`);
+	});
+
+	it('excludes partial files from building into their own files', () => {
+		buildSass(path.join(FIXTURES, 'partial'), {
+			outputDir: tempDir,
+		});
+
+		expect(fs.existsSync(path.join(tempSassDir, '_some-partial.css'))).toBe(
+			false
+		);
+	});
+
+	xit('builds with rtl support', () => {
```

Comment:
Just to double-check here, plan is to turn this back on via the code from liferay-r2 that's now in liferay-theme-tasks? Is that right?

As you probably saw, I imported liferay-js-theme-toolkit into this repo now, and all we have to do, I think, is officially expose the functions [in this file](https://github.com/liferay/liferay-frontend-projects/blob/22ab943c0663e04da755ad4cd88b433c105a8265/projects/js-themes-toolkit/packages/liferay-theme-tasks/lib/r2/liferay-r2/index.js) so that we can call them from `@liferay/npm-scripts`.

Seeing as I moved stuff around in the import, I was going to do a test release anyway just to make sure I didn't break anything.

Reply:
So I cut "test" releases yesterday (not really tests; just whatever small inconsequential stuff was on `master` since the last release), although the truth is you could already prior to the release use r2 hackily just by doing (something like):

```js
const {swap} = require('liferay-theme-tasks/lib/r2/liferay-r2');
```

If we want to use it, we should just [export it properly here](https://github.com/liferay/liferay-frontend-projects/blob/6fb97ab7a80dfd9d87a31c1d6c9dfb9a1c606b8a/projects/js-themes-toolkit/packages/liferay-theme-tasks/theme/index.js#L82-L84) to elevate it to "official/public" API. Only reason I didn't do this is because of your prior comment somewhere which leaves me uncertain of when we want to do it:

> I removed RTL support since we handle that via a filter in portal rather than building the files ahead of time. We can add this down the road, but figured its best to get the basics in first.

Reply:
Added https://github.com/liferay/liferay-frontend-projects/pull/301 which will mean you can do:

```js
const {r2} = require('liferay-theme-tasks');

r2.swap(...);
```